app/src/main/java/id/vouched/plugintest/MainActivity.kt

```diff
@@ -61,21 +90,35 @@ class MainActivity : AppCompatActivity(), VerificationListener {
     }
 
     override fun onRequestPermissionsResult(
-        requestCode: Int, permissions: Array<String?>,
+        requestCode: Int,
+        permissions: Array<String?>,
         grantResults: IntArray
     ) {
         super.onRequestPermissionsResult(requestCode, permissions, grantResults)
-        if (requestCode == CAMERA_PERMISSION_REQUEST) {
-            if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
-                cameraPermission?.grant(cameraPermission?.resources)
-            } else {
-                cameraPermission?.deny()
+        when (requestCode) {
+            REQUEST_FINE_LOCATION -> {
+                var allow = false
+                if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    // user has allowed this permission
+                    allow = true
+                }
+                if (geolocationCallback != null) {
+                    // call back to web chrome client
+                    geolocationCallback!!.invoke(geolocationOrigin, allow, false)
+                }
```

Comment:
due to `geolocationCallback` is defined as `Nullable` (`GeolocationPermissions.Callback?`) , you can just put in this way:  `geolocationCallback?.invoke(geolocationOrigin, allow, false)`

Reply:
If you wanted to do something when `geolocationCallback` is not null, you could do this:

```
geolocationCallback?.let {
  // doing something when it is not null
  it.invoke(geolocationOrigin, allow, false)
}
```

Reply:
Absolutely. This was a copy/paste from some java code I had written, ie it was Android Studios interpretation. Made those changes. Thanks